text/0317-third-party-dependency-management.md

```diff
@@ -0,0 +1,129 @@
+---
+rfc pr: [#319](https://github.com/aws/aws-cdk-rfcs/pull/319)
+tracking issue: https://github.com/aws/aws-cdk-rfcs/issues/317
+---
+
+# CDK third-party dependencies management
+
+The Cloud Development Kit depends on a large collection of open-source third-party libraries.
+While leveraging existing libraries that implement functionality needed by the project is obviously immensely useful,
+it also requires correctly managing those dependencies, in particular:
+
+1. Quickly reacting when a security vulnerability is discovered in a version of a dependency used by the project.
+
+2. Protecting ourselves against [supply chain attacks](https://en.wikipedia.org/wiki/Supply_chain_attack),
+  like [typosquatting](https://www.darkreading.com/vulnerabilities---threats/beware-the-package-typosquatting-supply-chain-attack/a/d-id/1340383),
```

Comment:
Is typosquatting in the same class as the other types of attacks?

Typo'ing is something our USERS would do, doesn't affect our own dependencies, right?

Reply:
We're as susceptible to typosquatting as our users (for example, we add a new dependency on `balacned-match`).

The ban on adding new dependencies would make us more resilient against this attack.